dist/cityManager.js

```diff
@@ -0,0 +1,54 @@
+class CityManager {
+    constructor() {
+        this.cityData = []
+    }
+    getCityData = async function (cityName) {
+        let cityInfo = await $.get(`/city/${cityName}`)
+        this.cityData.unshift(cityInfo)
+        return cityInfo
+    }
+
+    getDataFromDB = async function () {
+        let citiesData = await $.get('/cities')
+        citiesData.forEach(c => this.cityData.push(c))
+        return (this.cityData)
+    }
+
+    saveCity = async function (cityName) {
+        let CityToSave = this.cityData.find(c => c.name === cityName)
+        const newCity = await $.post('/city', CityToSave) 
+        let cityIndex = this.cityData.findIndex(c => c.name === cityName)
+        this.cityData.splice(cityIndex, 1)
```

Comment:
Instead of adding it at the beginning, try adding it at the exact same spot. The `splice` method can receive another argument  which adds something in the removed index.